src/bot.ts

```diff
@@ -815,16 +823,16 @@ export class DiscordBot {
                 return;
             }
             await Util.AsyncForEach(rooms, async (room) => {
-                const trySend = async () => intent.sendMessage(room, {
+                const trySend = async () => intent.sendEvent(room, {
                     body: result.body,
                     format: "org.matrix.custom.html",
                     formatted_body: result.formattedBody,
                     msgtype: result.msgtype,
                 });
-                const afterSend = async (re) => {
-                    this.lastEventIds[room] = re.event_id;
+                const afterSend = async (eventId) => {
+                    this.lastEventIds[room] = eventId;
                     const evt = new DbEvent();
-                    evt.MatrixId = `${re.event_id};${room}`;
+                    evt.MatrixId = `${eventId};${room}`;
                     evt.DiscordId = msg.id;
                     evt.ChannelId = msg.channel.id;
```

Comment:
It might be worth an idea instead of doing all this to write a join strategy that hooks into the appservice invite one and applies the userstore stuffs correctly